src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -124,10 +126,13 @@ private void parseTownCommand(Player player, String[] split) {
 
 			} else if (split[0].equalsIgnoreCase("list")) {
 
-				if (!TownyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_TOWN_LIST.getNode()))
+				if (!TownyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_NATION_LIST.getNode()))

```

Comment:
You should check for TOWNY_COMMAND_TOWN_LIST not TOWNY_COMMAND_NATION_LIST because this is the town list.